test-case/volume-basic-test.sh

```diff
@@ -96,8 +134,15 @@ do
                       func_error_exit "dmesg has errors!"
 done
 
-#clean up background aplay
-pkill -9 aplay || true
+if [[ "$SOF_ALSA_TOOL" = "alsa" ]]; then
```

Comment:
Extract the killing part of `func_error_exit()` above into a new function and invoke it here.